### PR TITLE
[Performance] Avoid double read file on parsing + create File object on creating StmtsAndTokens object

### PR DESCRIPTION
**Pull request by samsonasik**
@staabm @TomasVotruba this should improve performance with avoid double read file.

This PR read file once, use on both parsing and filling File object by using `parseString()` method.

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by staabm** [comment]
Any benchmark?

**Comment by samsonasik** [comment]
@staabm I tested run rector on rector-src, it seems faster:

**Before**

```
➜  rector-src git:(main) time bin/rector process --clear-cache
 2086/2086 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] Rector is done!                                                                                                   
                                                                                                                        

bin/rector process --clear-cache  219.56s user 10.26s system 713% cpu 32.191 total
```

**After**

```
➜  rector-src git:(dx-only-content) time bin/rector process --clear-cache
 2086/2086 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] Rector is done!                                                                                                   
                                                                                                                        

bin/rector process --clear-cache  216.03s user 8.79s system 715% cpu 31.410 total
```

It probably will more visible if the files has large content.

**Comment by TomasVotruba** [comment]
Not sure if 2 secs are worth it, but lets give it a go to cleanup code 👍

File: packages/NodeTypeResolver/NodeScopeAndMetadataDecorator.php [hunk 1]
```diff
@@ -36,9 +35,8 @@ public function __construct(
      * @param Stmt[] $stmts
      * @return Stmt[]
      */
-    public function decorateNodesFromFile(File|string $file, array $stmts): array
```
Comment (staabm) [inline packages/NodeTypeResolver/NodeScopeAndMetadataDecorator.php:39]: less union types is always a win :-)

**Pull request merged**